src/OrbitGl/ViewportTest.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include "Viewport.h"
+
+namespace orbit_gl {
+
+TEST(Viewport, ResizingAndDirty) {
+  Viewport viewport(100, 200);
+  EXPECT_TRUE(viewport.IsDirty());
+  viewport.ClearDirtyFlag();
+
+  // Test initial values
+  EXPECT_EQ(viewport.GetWidth(), 100);
+  EXPECT_EQ(viewport.GetHeight(), 200);
+  EXPECT_EQ(viewport.GetWorldExtents(), Vec2(100, 200));
+  EXPECT_EQ(viewport.GetWorldWidth(), 100.f);
+  EXPECT_EQ(viewport.GetWorldTopLeft(), Vec2(0.f, 0.f));
+
+  // Test: Resizing should not affect the world extents
+  viewport.Resize(1000, 2000);
+  EXPECT_TRUE(viewport.IsDirty());
+  viewport.ClearDirtyFlag();
+
+  EXPECT_EQ(viewport.GetWidth(), 1000);
+  EXPECT_EQ(viewport.GetHeight(), 2000);
+  EXPECT_EQ(viewport.GetWorldWidth(), 100.f);
+  EXPECT_EQ(viewport.GetWorldHeight(), 200.f);
+  EXPECT_EQ(viewport.GetWorldExtents(), Vec2(100, 200));
+  EXPECT_EQ(viewport.GetWorldTopLeft(), Vec2(0.f, 0.f));
+
+  // Test: Changing the world width / height should not affect screen extents
+  viewport.SetWorldWidth(500.f);
+  EXPECT_TRUE(viewport.IsDirty());
+  viewport.ClearDirtyFlag();
+
+  viewport.SetWorldHeight(600.f);
+  EXPECT_TRUE(viewport.IsDirty());
+  viewport.ClearDirtyFlag();
+
+  EXPECT_EQ(viewport.GetWidth(), 1000);
+  EXPECT_EQ(viewport.GetHeight(), 2000);
+  EXPECT_EQ(viewport.GetWorldWidth(), 500.f);
+  EXPECT_EQ(viewport.GetWorldHeight(), 600.f);
+  EXPECT_EQ(viewport.GetWorldTopLeft(), Vec2(0.f, 0.f));
+  EXPECT_EQ(viewport.GetWorldExtents(), Vec2(100, 200));
```

Comment:
I agree with this being not super easy to grasp at the first glance. Maybe consider adding a more detailed comment on the method (IDE's should show the comments also on the caller).

Reply:
Good point. I was thinking the same and was considering to rename `WorldWidth` to `WorldVisibleWidth` or something, and decided against it to make it easier to review (as the naming doesn't change so much). But I think I'll actually add the renaming to this PR.

Reply:
I renamed all methods:

VisibleWorld[Width/Height]: Size of the world visible on the screen
WorldExtents: Max values of the current world
WorldMin: Min values of the current world (*technically*, min and max are swapped for y because all values we use are negative, but conceptually, I think I'll stick with this notion of min / max)
ScreenSize: Size of the screen in pixels

I added some comments above each method block, hope that makes the usage clearer.